src/constructors.jl

```diff
@@ -66,174 +66,38 @@ function Base.Array(chn::Chains,
         showall=false,
         sorted=false
     )
-    sections = _clean_sections(chn, sections)
-    sections = sections isa AbstractArray ? sections : [sections]
-    sections = showall ? [] : sections
-    section_list = length(sections) == 0 ?
-        sort_sections(chn) :
-        sections
+    sections = showall ? keys(chn.name_map) : sections
+    chn = Chains(chn, sections)
 
-    # If we actually have missing values, we can't remove
-    # Union{Missing}.
     remove_missing_union = remove_missing_union ?
         all(x -> !ismissing(x), chn.value) :
         remove_missing_union
 
-    d, p, c = size(chn.value.data)
-
-    local b
-    if append_chains
-        first_parameter = true
-        for section in section_list
-            for par in chn.name_map[section]
-                x = get(chn, Symbol(par))
-                d, c = size(x[Symbol(par)])
-                if first_parameter
-                    if remove_missing_union
-                        b = reshape(convert(Array{Float64}, x[Symbol(par)]), d*c)[:, 1]
-                    else
-                        b = reshape(x[Symbol(par)], d*c)[:, 1]
-                    end
-                    p == 1 && (b = reshape(b, size(b, 1)))
-                    first_parameter = false
-                else
-                    if remove_missing_union
-                        b = hcat(b, reshape(convert(Array{Float64}, x[Symbol(par)]), d*c)[:, 1])
-                    else
-                        b = hcat(b, reshape(x[Symbol(par)], d*c)[:, 1])
-                    end
-                end
-            end
+    arr = if append_chains
+        if remove_missing_union
```

Comment:
Are the types not made more concrete by the `map(identity, ...)` calls below, so that `remove_missing_union` can be dropped here?